File: 0001-two-sum/solution.rs

pub struct Solution;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        
    }
}

// --- Test Runner ---
fn parse_int_vec(s: &str) -> Vec<i32> {
    let s = s.trim();
    if s == "[]" { return vec![]; }
    let s = s.trim_start_matches('[').trim_end_matches(']');
    s.split(',').map(|x| x.trim().parse().unwrap()).collect()
}

fn main() {
    use std::fs;
    let content = fs::read_to_string("testcases.txt").expect("testcases.txt not found");
    let blocks: Vec<&str> = content.split("\n\n").filter(|b| !b.trim().is_empty()).collect();
    
    for (i, block) in blocks.iter().enumerate() {
        println!("Test {}:", i + 1);
        
        let raw_lines: Vec<&str> = block.lines().collect();
        let mut lines = Vec::new();
        let mut expected = None;
        let mut is_input = false;
        
        for line in raw_lines {
            if line.starts_with("input:") {
                is_input = true;
            } else if line.starts_with("output:") {
                is_input = false;
                let inline = line.strip_prefix("output:").unwrap().trim();
                if !inline.is_empty() {
                    expected = Some(inline.to_string());
                }
            } else if is_input {
                lines.push(line);
            } else if !is_input && expected.is_none() && !line.trim().is_empty() {
                expected = Some(line.trim().to_string());
            }
        }
        
        if lines.len() < 2 { continue; }
        
        let arg0 = parse_int_vec(lines[0]);
        let arg1 = lines[1].trim().parse::<i32>().unwrap();
        
        let result = Solution::two_sum(arg0, arg1);
        // 打印输入
        println!("  Input: {:?}", &lines[0..2.min(lines.len())]);
        
        println!("  Got: {}", format!("{:?}", result));
        
        if let Some(exp_str) = expected {
            let exp = parse_int_vec(&exp_str);
            println!("  Exp: {:?}", exp);
            if result == exp {
                println!("  ✅ Passed");
            } else {
                println!("  ❌ Failed");
            }
        } else {
            println!("  (No expected output)");
        }
        println!("--------------------");
    }
}

File: 0001-two-sum/solution.go
package main

import (
    "encoding/json"
    "fmt"
    "os"
    "strings"
)


func twoSum(nums []int, target int) []int {
    
}

// --- Test Runner ---
func main() {
    data, err := os.ReadFile("testcases.txt")
    if err != nil {
        fmt.Println("testcases.txt not found")
        return
    }
    
    content := string(data)
    content = strings.ReplaceAll(content, "\r\n", "\n")
    blocks := strings.Split(content, "\n\n")
    
    testNum := 0
    for _, block := range blocks {
        block = strings.TrimSpace(block)
        if block == "" {
            continue
        }
        testNum++
        
        rawLines := strings.Split(block, "\n")
        var lines []string
        var expected string
        isInput := false
        
        for _, line := range rawLines {
            if strings.HasPrefix(line, "input:") {
                isInput = true
            } else if strings.HasPrefix(line, "output:") {
                isInput = false
                inline := strings.TrimSpace(strings.TrimPrefix(line, "output:"))
                if inline != "" {
                    expected = inline
                }
            } else if isInput {
                lines = append(lines, strings.TrimSpace(line))
            } else if !isInput && expected == "" && strings.TrimSpace(line) != "" {
                expected = strings.TrimSpace(line)
            }
        }
        
        if len(lines) < 2 {
            continue
        }
        
        fmt.Printf("Test %d:\n", testNum)
        
        var arg0 []int
        json.Unmarshal([]byte(lines[0]), &arg0)
        var arg1 int
        json.Unmarshal([]byte(lines[1]), &arg1)
        
        result := twoSum(arg0, arg1)
        // 打印输入
        inputSlice := lines
        if len(lines) > 2 {
            inputSlice = lines[:2]
        }
        fmt.Printf("  Input: %v\n", inputSlice)
        
        fmt.Printf("  Got: %v\n", result)
        
        if expected != "" {
            var exp []int
            json.Unmarshal([]byte(expected), &exp)
            fmt.Printf("  Exp: %v\n", exp)
            
            if fmt.Sprint(result) == fmt.Sprint(exp) {
                fmt.Println("  ✅ Passed")
            } else {
                fmt.Println("  ❌ Failed")
            }
        } else {
            fmt.Println("  (No expected output)")
        }
        fmt.Println("--------------------")
    }
}
